### ISSUE TITLE
Broken URL links in CSS

### ISSUE THREAD

**Issue by lokenx** [fizker/minifier#22]
Maybe it's intended but I ran this on a CSS file with the below `background-image` and it converted it to one that doesn't reference the correct file.

```cs
/* original */

body {
  background-image: url('lib/background.jpg');
}

/* minified */

body {
  background-image: url(../lib/background.jpg);
}
```

**Comment by fizker** [comment]
The expected behaviour depends on the context. As an example, let's say that your project have the following file-structure:

```
- a.css
- build/
- css/
  - b.css
- lib/
  - background.jpg
```

where

- `a.css` contains the snippet you mentioned.
- `b.css` contains `@import url(../a.css);`

In this scenario, I would expect your output in the following scenarios:

- You minify `b.css` and leave the result next to the original
- You minify `a.css` and tell the tool to put the result into the `build/` folder.

I would consider it a bug in the following scenarios:

- You minify `a.css` and leave the result next to the original.
- You minify `b.css` and tell the tool to put the result next to `a.css`.

---

There are a ton of alternative scenarios as well, so I really need more context to get any further with this issue.

Is your project on GitHub or otherwise publicly available? That would make it real easy for me to dig around and see what makes it tick :).

**Comment by lokenx** [comment]
Not public unfortunately but the relevant structure is below:

```
- lib/
  - background.jpg
- dist/
  - lib/background.jpg
  - style.min.css
- style.css
```

So I copied the `lib/` folder into `dist/` and then ran `minify --output dist/style.css style.css` to copy the css and minify in one step. Is it because it was being minified down a folder it added the `../` to the start of the link because it thought it was referencing that `lib` instead of the one in `dist`?

At least thats my best guess!

**Comment by fizker** [comment]
Yep. Because the file that you are minifying _is indeed_ referencing the image in the root `lib/` folder.

Currently, the simplest workaround for you would be to output the minified next to the original, and then move it into the `dist/` folder.

It might make sense for an option to set the root folder used for the url-resolving. Right now it is calculating the root based on the input and output folders, which trips your build up. I will sleep on it, and make a proper decision tomorrow. It is getting too late for me to get anything done tonight anyway :).

**Comment by lokenx** [comment]
Ah OK, I didn't know it was doing any smarts like that, I thought it was just a "dumb" minification. I don't think it's a bug per-se, but maybe an update of the README or mention somewhere that links are relative to the folders or something?

Let me know your thoughts and I can submit a PR if wanted!

**Comment by fizker** [comment]
A PR would be welcome. It is a consequence of the feature which inlines `@import` statements; in those scenarios, any relative urls need to be updated to reflect the new position, or they no longer point to the same location.

**Comment by fizker** [comment]
BTW, from the readme: `It reworks URLs in CSS from the original location to the output location`.

This is the thing that got you confused. It is apparently not as good a statement as I thought. Perhaps an example or different wording would have avoided the confusion.

**Comment by lokenx** [comment]
Yeah I saw that while reading it now, definitely isn't 100% clear, at least to me! I'll have a look on the weekend and see if can come up with better wording or somewhere else to mention it!

**Comment by lokenx** [comment]
Was addressed in #23
